src/components/layout/Layout.tsx

```diff
@@ -2,29 +2,32 @@ import { NavLink } from 'react-router-dom';
 import { ChangeEvent, useState } from 'react'
 import style from './Layout.module.scss';
 import { getCookie } from 'typescript-cookie';
+import { someSlice } from '../../store/inputReducer';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { useEffect } from 'react';
+import { getAllCharacters } from '../../services/Api';
 
 
-interface LayoutProps {
-  setIsModalVisible: (boolean: boolean) => void
-  searchedCharacter: (charName:string) => void
-}
-
+const Layout:React.FC = () => {
 
+  const {inputText, isModalVisible, allCharacters} = useAppSelector(state => state.inputReducer);
+  const {changeInputText, openModal, defineSearchedCharacter, fetchAllCharacters, clearSearchedCharacter} = someSlice.actions;
+  const dispatch = useAppDispatch();
 
-const Layout:React.FC<LayoutProps> = ({setIsModalVisible, searchedCharacter}) => {
-
-  const [inputValue, setInputValue] = useState<string>('');
+  const inputedCharacter = allCharacters.find((chrs) => chrs.name.toLowerCase() === inputText.toLowerCase())
```

Comment:
if you'll open your app directly at random chars page, where will you get your chars list?

Reply:
tip: consider fetch the character list at the random char page